Makefile

```diff
@@ -55,6 +55,27 @@ $(OUT)/emulate.o: CFLAGS += -fno-gcse -fno-crossjumping
 endif
 endif
 
+ENABLE_GDBSTUB ?= 1
+ifeq ("$(ENABLE_GDBSTUB)", "1")
+MINI_GDBSTUB_OUT = $(abspath $(OUT)/mini-gdbstub)
+GDBSTUB_COMM = 127.0.0.1:1234
+LIB_GDBSTUB += $(MINI_GDBSTUB_OUT)/libgdbstub.a
+gdb_test: $(BIN)
+	./build/rv32emu --gdbstub ./build/puzzle.elf &
```

Comment:
1. Move GDB scripts to directory `tests`.
2. Rename `gdb_test` to `gdbstub-test`
3. Check the results of running GDB script. e.g., check the value of pc register.

Reply:
You can use the `jobs` command to the get PIDs and then use `wait` with the PID. Check the manual about builtin commands in bash.

Reply:
I will write a shell script for the check because I think it will become a mess to write those in Makefile directly. Also, we may have more complicated test patterns in the future.